### PR TITLE
[Feature-Request] Front - v1.0 (back not fully plugged)

### PR DESCRIPTION
**Pull request by spicytelescope**
This PR aims to present the v1.0 of the front, containing all of the components as design from the [Figma sketch](https://www.figma.com/file/IGFPnJcCLXqHKNUoFAc41X/Library?node-id=0%3A1), a complete working integration with 3D rendering of the flag for a better experience, and an authentification handling (with route-guarding on).
Quick reminder : the front has been built with a Nuxt environnement (requiring [tailwindcss](https://tailwindcss.com/) styling).

> **_WARNING :_** these components have been build to be responsive, but some usecase have not been treated yet. Do not hesitate to make any issues report, or PR if a components doesn't seems to display correctly under specifics breakpoints. You can also contact the front team !
> Furthermore, the back has yet to be plugged as some problems with the Fouloscopie SDK are still unresolved.
> ℹ️ For further informations about what is yet to come, please look at the **Todo** section.

# New

## 3D Rendering

- A 3D scene has been added to represent the flag, with the `ThreeJS` library
- To have a better UX, a windy effect have been added to the scene

## Assets

- `orbitControl.js` is an asset needed to handle the control of the 3D camera, dealing with `Threejs` requirements

## Styling

- Added responsiveness on almost every components, especially the **header**

### Atoms

- **AppCheckbox** : a custom checkbox input
- **AppStep** : a part of the `AppStepLine` component (see **Molecules** section for more details), represents a step of a basic stepper component
- [UPDATED] **AppInputFrame** : now contains all the features such as label, errorMessage

### Molecules

- **AppStepLine** : a custom stepper component, can be inserted into modals
- [REMOVED] **LabeledInputFrame** : deleted because merged with a custom text input to create the `AppInputFrame` component
### Organism

- **AppAlert** : the navbar of micronation, showing differents buttons given the authentification state of the user
- **AppCard** : the navbar of micronation, showing differents buttons given the authentification state of the user

## Pages

- **`about.vue`** : contains the credits and a brief description about the experience
- **`edit.vue`** : user can edit his pixel on the flag, and see in detail the current composition of it (using canvas and dom manipulation through vanilla javascript)

## Back-end

- **Authentification** : implemanted route guarding with the help of the Fouloscopie SDK, only the `/edit` endpoint isn't reachable without the proper `userToken` (verified by the SDK)
- **Micronation API** : integrated the first API Call through `fetch`

# Todo

### Responsive design

- [ ] Make sure that every components under the usual breakpoint are UI-responsive

### Pages

- [ ] Fill out the about page with the right text, credits, etc
- [ ] Add transitions and modals (**AppAlert**) to improve the UX (use_case: a user changes a pixel, modals to get the confirmation that the modificatio has correctly reached the server)

### Back-end

- [ ] Solve the Fouloscopie SDK problems **[in-progress]**
- [ ] Fix the fetch calls, regarding the right headers (Authorization)
- [ ] Fix the functions to fetch a pixel and create the flag from the array of colors coming from the API

### CODE REVIEW

File: back/library/database/DatabaseModule.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            ],
+            exports: ['DATABASE_CLIENT']
+        };
+    }
+};
+DatabaseModule = DatabaseModule_1 = __decorate([
+    common_1.Module({})
+], DatabaseModule);
+exports.DatabaseModule = DatabaseModule;
+//# sourceMappingURL=DatabaseModule.js.map
```
Comment (ashpieboop) [inline back/library/database/DatabaseModule.js:36]: Please prune generated files from the back/ module (js, maps, ts declaration, tsconfig.tsbuildinfo) from your PR.

**Review by ashpieboop — CHANGES_REQUESTED** [review]
Please prune generated files of the back/ module (js, maps, ts declaration, tsconfig.tsbuildinfo) from your PR.

**Comment by spicytelescope** [comment]
@ashpieboop Pushed 3 more commits, is this solving the issue ? 
And is the `.gitignore` file in `back/library` correctly setup ? 

*Feel free to make any changes necessary !!*

**Comment by ashpieboop** [comment]
I didn't have write access to your PR's branch so I made a PR on your PR : https://github.com/spicytelescope/micronation/pull/1

**Comment by spicytelescope** [comment]
Closing requests due to more recent version fixings bugs and improving the back-plug.

**Comment by spicytelescope** [comment]
Closing requests due to more recent version fixings bugs and improving the back-plug.

**Pull request closed without merging**